src/contracts/atlas/Escrow.sol

```diff
@@ -169,9 +168,10 @@ abstract contract Escrow is AtlETH {
             }
         }
 
+        // If we reach this point, the solver call did not execute successfully.
         ctx.solverOutcome = uint24(result);
 
-        _releaseSolverLock(solverOp, gasWaterMark, result, false, !prevalidated);
+        _handleSolverAccounting(solverOp, gasWaterMark, result, !prevalidated);
```

Comment:
Auditors recommend call this (was previously _releaseSolverLock) anywhere there is an early return and we want to  ensure the solver covers some/all of the gas if its their fault. Audit issue: https://github.com/spearbit-audits/review-fastlane/issues/78

But in the `allowsTrustedOpHash` block above (L145) we return without calling `_handleSolverAccounting()`.

Should we be calling it there and making sure `result` reflects that this is the bundler's fault so gasUsed is added to writeoffs?

Reply:
Yes this change makes sense to me

Reply:
Added it here https://github.com/FastLane-Labs/atlas/pull/271/commits/ae42860f2e04665caf0e142b2868e0b500c9d206

I don't think the `result` var needs to be adjusted - it should already have the info to apportion blame in `_handleSolverAccounting()`.

Also added an event emit just before the `return 0` as that seems to also be missing.

@thogard785 pls double check this change in particular